idseq_dag/util/m8.py

```diff
@@ -181,21 +183,49 @@ def accumulate(hits, accession_id):
                 # provided by other accessions. This occurs a lot and
                 # handling it in this way seems to work well.
                 continue
-            hits[level][taxid_at_level] = accession_id
+            accession_list = hits[level].get(taxid_at_level, []) + [accession_id]
```

Comment:
I see, here you are remembering all the accession ids that contributed to that taxid, so you can choose the taxid with the largest number of accession ids that contributed to it.

Reply:
On a separate note, the blacklist handling here seems suboptimal.  In the event a read maps to the blacklist but also to something else, we probably don't want to discard the read.  Rather, we should probably retain the mapping to the real pathogen.  Perhaps all that is needed here is that you change "break" to "continue".